File: commercehub-wishlist/src/main/java/com/commercehub/wishlist/service/WishlistService.java
package com.commercehub.wishlist.service;

import com.commercehub.wishlist.dto.WishlistItemResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.UUID;

public interface WishlistService {

    Page<WishlistItemResponse> getWishlist(UUID userId, Pageable pageable);

    WishlistItemResponse addToWishlist(UUID userId, UUID productId);

    void removeFromWishlist(UUID userId, UUID productId);
}

File: commercehub-frontend/src/features/wishlist/services/wishlistService.ts
import { apiClient } from '@/api/client';
import type { ApiResponse, PaginatedResponse } from '@/types/api';
import type { WishlistItemResponse, WishlistResponse } from '../types';

export const wishlistService = {
  getWishlist: async (): Promise<WishlistResponse> => {
    const response = await apiClient.get<PaginatedResponse<WishlistItemResponse>>('/wishlist');
    return response.data.data;
  },

  addToWishlist: async (productId: string): Promise<WishlistItemResponse> => {
    const response = await apiClient.post<ApiResponse<WishlistItemResponse>>(`/wishlist/${productId}`);
    return response.data.data;
  },

  removeFromWishlist: async (productId: string): Promise<void> => {
    await apiClient.delete(`/wishlist/${productId}`);
  },
};
